lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -373,9 +374,53 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 		$with_theme_supports = new WP_Theme_JSON_Gutenberg( $theme_support_data );
 
 		if ( $options['with_block_style_variations'] ) {
+			// Absorb block style variations that were registered with a style object.
 			$block_style_variations_data = WP_Theme_JSON_Gutenberg::get_from_block_styles_registry();
 			$with_block_style_variations = new WP_Theme_JSON_Gutenberg( $block_style_variations_data );
 			$with_theme_supports->merge( $with_block_style_variations );
+
+			// Resolve shared block style variations that were bundled in the
+			// theme via standalone theme.json files.
+			$shared_block_style_variations = static::get_style_variations( '/block-styles' );
```

Comment:
For the moment, the shared block style variations are being looked for in a `block-styles` directory within the theme. As with theme style variations under `styles`, this takes into account child theme variations overriding a parent's definition.

Is there a better location to store these block style variations? It seemed keeping them separate from theme style variations made sense to limit further updates to ensure block style variations weren't accidentally included with theme styles.

Reply:
I think it needs to be a different directory than the existing styles, but the naming is trickier.